protocol/directory.go

```diff
@@ -0,0 +1,109 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+type ConiksUserPolicies struct {
+	Username               string
+	AllowUnsignedKeychange bool
+	AllowPublicLookup      bool
+}
+
+type ConiksDirectory struct {
+	pad      *merkletree.PAD
+	useTBs   bool
+	tbs      map[string]*merkletree.TemporaryBinding
+	policies merkletree.Policies
+}
+
+func InitDirectory(epDeadline merkletree.TimeStamp, vrfKey *vrf.PrivateKey,
+	signKey sign.PrivateKey, dirSize uint64,
+	useTBs bool, capacity uint64) *ConiksDirectory {
```

Comment:
Or we could even drop the capacity completely (as it is "just" an optimization).